src/lib/jsonParseV2/parseRegistration.cpp

```diff
@@ -0,0 +1,367 @@
+      return false;
+    }
+
+    providerP->supportedForwardingMode = supportedForwardingMode.GetString();
+    
+    //
+    // Valid strings: ngsiV2, legacy, all
+    //
+    if ((providerP->supportedForwardingMode != "ngsiV2") &&  // FIXME PR: what are the exact values?
+        (providerP->supportedForwardingMode != "legacy") &&
+        (providerP->supportedForwardingMode != "all"))
+    {
+      *errorStringP = "invalid value of /supportedForwardingMode/";
+      return false;
+    }
+  }
+  else
+  {
+    providerP->supportedForwardingMode = "legacy";  // FIXME PR: what is the default value?
+  }
+
+  return true;
+}
+
+
+
+/* ****************************************************************************
+*
+* parseRegistration -
+*/
+std::string parseRegistration(ConnectionInfo* ciP, ngsiv2::Registration* regP)
+{
+  std::string          errorString;
+  rapidjson::Document  document;
+
+  document.Parse(ciP->payload);
+
+  if (document.HasParseError())
+  {
+    return badInput(ciP, ERROR_DESC_PARSE);
+  }
+
+  if (!document.IsObject())
+  {
+    return badInput(ciP, ERROR_DESC_BAD_REQUEST_NOT_A_JSON_OBJECT);
+  }
+
+  if (document.ObjectEmpty())
+  {
+    return badInput(ciP, ERROR_DESC_BAD_REQUEST_EMPTY_PAYLOAD);
+  }
+
+
+  //
+  // Extracting field "description"
+  //
+  Opt<std::string> description = getStringOpt(document, "description");
+
+  if (!description.ok())
+  {
+    return badInput(ciP, "invalid description");
+  }
+  else if (description.given)
+  {
+    regP->description = description.value;
+    if (regP->description.length() > MAX_DESCRIPTION_LENGTH)
+    {
+      return badInput(ciP, "max description length exceeded");
+    }
+
+    if (forbiddenChars(regP->description.c_str()))
+    {
+      return badInput(ciP, "forbidden characters in description");
+    }
+  }
+
+
+  //
+  // Extracting field "dataProvided"
+  //
+  if (!document.HasMember("dataProvided"))
+  {
+    return badInput(ciP, "the field /dataProvided/ is missing in payload");
+  }
+
+  const rapidjson::Value& dataProvided = document["dataProvided"];
+
+  if (dataProvidedParse(ciP, &regP->dataProvided, dataProvided, &errorString) == false)
+  {
+    return badInput(ciP, errorString);
+  }
+
+
+  //
+  // Extracting field "provider"
+  //
+  if (!document.HasMember("provider"))
+  {
+    return badInput(ciP, "the field /provider/ is missing in payload");
+  }
+
+  const rapidjson::Value& provider = document["provider"];
+
+  if (providerParse(ciP, &regP->provider, provider, &errorString) == false)
+  {
+    return badInput(ciP, errorString);
+  }
+
+
+  //
+  // Extracting field "expires"
+  //
+  if (document.HasMember("expires"))
+  {
+    const rapidjson::Value& expires = document["expires"];
+
+    if (!expires.IsString())
+    {
+      return badInput(ciP, "/expires/ must be a JSON string");
+    }
+
+    regP->expires = parse8601Time(expires.GetString());
+  }
+  else
+  {
+    // FIXME PR: default expiration?
```

Comment:
Not sure of understanding what do you mean by this FIXME PR comment, sorry...

Reply:
This one is easier ... 
In ngsiv1, default expiration was 24 hours (, I think). What is the default expiration in APIv2 for registrations ?

Reply:
According to http://telefonicaid.github.io/fiware-orion/api/v2/latest/

> `expires`: Registration expiration date in ISO8601 format. Permanent registrations must omit this field.

So if a registration doesn't inlclude the field, I understand it will not use expiration. In other words, the 24 hours limits applies only to NGSIv1 registrations.

Reply:
ok

Reply:
Fixed in 5686ca2  (`expires` set to "MAXLONGLONG")